perses/rjmc/geometry.py

```diff
@@ -525,34 +560,37 @@ def _propose_bond(self, bond, beta):
         """
         Bond length proposal
         """
-        k_eq = bond.type.k*units.kilocalories_per_mole/(units.angstrom**2)
-        r0 = bond.type.req*units.angstrom
-        sigma = beta*2.0/np.sqrt(2.0*k_eq/k_eq.unit)
-        r = sigma/sigma.unit*np.random.random()*r0.unit + r0
+        r0 = bond.type.req
+        k = bond.type.k
+        sigma_r = units.sqrt(1.0/(beta*k))
+        r = sigma_r/sigma_r.unit*np.random.random()*units.angstrom + r0
```

Comment:
Oops, this is what I was doing before. I think I misunderstood a previous comment, then.